docs/root/configuration/http/http_filters/wasm_filter.rst

```diff
@@ -21,20 +21,10 @@ Example configuration
 
 Example filter configuration:
 
-.. code-block:: yaml
-
-  name: envoy.filters.http.wasm
-  typed_config:
-    "@type": type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
-    config:
-      config:
-        name: "my_plugin"
-        vm_config:
-          runtime: "envoy.wasm.runtime.v8"
-          code:
-            local:
-              filename: "/etc/envoy_filter_http_wasm_example.wasm"
-          allow_precompiled: true
-
+.. literalinclude:: ../../../start/sandboxes/_include/wasm-cc/envoy.yaml
+    :language: yaml
+    :lines: 24-43
```

Comment:
Are those numbers correct? It looks that this would start at `cluster: web_service` and end at `name: web_service`, so a few lines too much on each side. I think it should be lines 26-38?

Reply:
rendered docs are here https://storage.googleapis.com/envoy-pr/02d2153/docs/configuration/http/http_filters/wasm_filter.html#example-configuration - it follows the same convention that other similar docs have of showing the last part of previous config stanza and first part of next to give some context

Reply:
I clicked around in the documentation looking at the examples for other HTTP filters, and I couldn't find any that included unrelated parts of the configuration.

Do you have any examples that follow this convention? Quite frankly, I find the linked snippet extremelly confusing.

Reply:
- https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/configuration-static#listeners
- https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/configuration-dynamic-filesystem#node
- https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/configuration-dynamic-filesystem#dynamic-resources
- https://www.envoyproxy.io/docs/envoy/latest/start/quick-start/configuration-dynamic-control-plane#dynamic-resources

...and a few more `git grep linenos` shows all of the partially literally included configs

Reply:
actually `git grep ":lines:"` is the more accurate search - most have line numbers included - would it help if i add the line numbers to this config ?

Reply:
It think it makes sense to add the line numbers to match the other example, but it doesn't really address my issue.

The example from this PR shows:
```yaml
                  cluster: web_service

          http_filters:
          - name: envoy.filters.http.wasm
            typed_config:
              "@type": type.googleapis.com/envoy.extensions.filters.http.wasm.v3.Wasm
              config:
                name: "my_plugin"
                root_id: "my_root_id"
                vm_config:
                  runtime: "envoy.wasm.runtime.v8"
                  vm_id: "my_vm_id"
                  code:
                    local:
                      filename: "lib/envoy_filter_http_wasm_example.wasm"
          - name: envoy.filters.http.router
            typed_config: {}

  clusters:
  - name: web_service
```

I agree that having complete `http_filters` block and highlighting only `envoy.filters.http.wasm` filter is useful.

However, I see absolutely zero value in including completely unrelated surrounding (first and the last 2 lines).


In some of the other examples, it's downright confusing. For example, this documentation for `node:`:

```yaml
node:
  cluster: test-cluster
  id: test-id

dynamic_resources:
```

What's the value of adding `dynamic_resources:`?

Reply:
im afraid i disagree about this

when first starting with envoy config having bare snippets doesnt tell you where in the config the snippet should go

personally i found this an absolute nightmare - constantly having to go and try and find other snippets that might elucidate the structure

the point is *dont assume that someone knows where `http_filters` is placed

its definitely beyond the scope of this PR to change that approach - this PR  fixes what is currently a broken example snippet ie #15558

Reply:
...and to add validation to ensure the config actually works

Reply:
I disagree on the readability, but I agree that this is beyond this PR, sorry for sidetracking. Let's move on.

Reply:
> the point is _dont assume that someone knows where `http_filters` is placed_

I agree, but I don't see how having random `cluster: web_service` helps with that.

Reply:
its just a visual cue that there is more to this config that will show you where it sits - its not perfect - but on balance after working through a lot of config it was the best idea i came up with